corehq/apps/es/users.py

```diff
@@ -64,13 +64,17 @@ def show_only_inactive(self):
         return query.is_active(False)
 
 
-def domain(domain, allow_enterprise=False):
+def domain(domain, allow_enterprise=False, allow_enterprise_controlled_domains=False):
+    from corehq.apps.enterprise.models import EnterprisePermissions
     domain_list = [domain]
     if allow_enterprise:
-        from corehq.apps.enterprise.models import EnterprisePermissions
         config = EnterprisePermissions.get_by_domain(domain)
         if config.is_enabled and domain in config.domains:
             domain_list.append(config.source_domain)
+    if allow_enterprise_controlled_domains:
+        config = EnterprisePermissions.get_by_domain(domain)
+        if config.is_enabled:
```

Comment:
I think this should also check that `domain == config.source_domain`. As is, if you're in a county, this will allow you to see all the other counties, but you really want all counties to be available only if you're in the state domain, right?

Reply:
aha, thank you. I got hung up on how `get_by_domain` works for both source & controlling domains and missed that `get_domains` only deals with source domains.